### DISCUSSION TITLE
Song keeps repeating

### DISCUSSION THREAD

**Discussion by ChandigarIT** [AzuraCast/AzuraCast#7374] [Q&A]
Is there a way to prevent songs from constantly repeating? It's a different one each day, today is Sweet Little 16 by Chuck Berry. See screenshot of timeline.

Thanks.

![image](https://github.com/user-attachments/assets/82a38cc2-6656-4b82-a26c-60999ef5a6f4)

**Comment by Heimatradio** [comment]
Maybe playlist 1 is not set to random

**Reply by ChandigarIT** [reply]
It was Shuffled, I've changed it to Random to see what happens. Many thanks.

**Comment by SpeedyCharly** [comment]
You must also set the No Repeat widget to at least 60 minutes...

**Reply by ChandigarIT** [reply]
Where is the No Repeat widget? Is this not the "Avoid Duplicate Artists/Titles"?

**Reply by SpeedyCharly** [reply]
Edit Profile > AutoDj ; bottom of page
![repeat](https://github.com/user-attachments/assets/912810b0-c830-4f65-8184-2e4aaf77433a)
You have to set a time limit, 60 mins being fine for most cases.

Then ensure that No repeat is enabled in the playlist (as you have done). "Avoid Duplicate Artists/Titles" in the playlist is the enable/disable switch.

**Reply by ChandigarIT** [reply]
Found it, already set at 120.

**Comment by ChandigarIT** [comment]
It's still repeating songs, 
![image](https://github.com/user-attachments/assets/7ccec3ca-83c8-4ecb-92ff-93a6a5467b75)


Set to "random" and "avoid duplicates"
![image](https://github.com/user-attachments/assets/0b927b4a-3aa3-41b2-a719-21a23de34706)

Also "play only one song" is NOT checked
![image](https://github.com/user-attachments/assets/d61c5cb2-fa83-4ef8-908b-d0ad3cb848aa)

but it also keeps playing songs from the default playlist.

If the default playlist interrupts the song that is playing from Playlist 1, will it restart again?

**Reply by SpeedyCharly** [reply]
Default playlist shouldn't interrupt a song (unless specifically set to do so, which would be wrong in this case). They should play in sequence as set in your queue. 

As I see it here in your settings, AutoDj should be picking songs randomly from either playlist equally if playlist weight is set equal or favor the playlist with heavier weight and play them in sequence

**Comment by gAlleb** [comment]
To avoid repetitions of the song you have to select "shuffled mode". In "random" mode tracks are picked randomly - thus repetitions are possible. And restart Azuracast just in case.

**Reply by SpeedyCharly** [reply]
Sorry, I don't believe that is the case.  I have been in random mode since forever and my no repeat settings work just fine.  Yes, in random mode tracks are picked randomly but when played, a tag is appended to them stating last time played and if it gets picked again within your no repeat set time (typically 60 mins) it will be rejected and another selection will be sent to the queue.

Also, and obviously, the amount of tracks in your playlist will affect the frequency of repetition... Can't avoid repetition if you're only shuffling 10 items...

**Reply by gAlleb** [reply]
`no repeat` option with `random playlist` can mimic `shuffled playlist` behavior to a certain point (because it can repeat `the same song` after the period of time set in the `no repeat` option has passed) and exceed it in the case of non-repeating an artist or title. 

However, if we're talking about a certain song - then `shuffled playlist` will play it only once during its round. If its round consists of 3 tracks - then songs will be repeated every time for sure. In this case I would consider it a user error to expect a song in a small playlist to be non-repetitive. Adding `no repeat` option, however should solve this issue even in this case.

**Reply by SpeedyCharly** [reply]
IMHO shuffled playlists are useful only if you're streaming from a single playlist where tracks get shuffled then played sequentially. In such case, the no repeat is of no use as no track will repeat for the duration of the list.
 In my station we shuffle something like 5000 tracks set in 10 different playlists.  Therefore I have to pick randomly from each playlist, all of which have the no repeat activated.  I also have to play with playlist weight and per minute settings in order to emulate a proper clockwheel which, sadly enough, doesn't yet exist in Azuracast...

**Comment by ChandigarIT** [comment]
I found out that someone had set the default to play every so many minutes, and I suspect that this was interrupting the song on Playlist1. I've changed that to General Rotation, and it seems to have worked, so far. So the next suggestion is that when a track is interrupted, it is either continued where it left off, or abandoned altogether, rather than start from the beginning again.

**Reply by SpeedyCharly** [reply]
Yes General Rotation should be right in this case.

If ever a song is interrupted for whatever reason, it is my understanding that it is dropped altogether from the queue

**Reply by ChandigarIT** [reply]
Definitely NOT dropped, according to the Song Timeline. Unless there is something somewhere in Azuracast settings where you can change this.

**Reply by SpeedyCharly** [reply]
Can't understand why your setup wouldn't behave anything like mine...
1st - Songs shouldn't be interrupted on change of playlist unless the "Interrupt other song" switch is clicked on.
2nd - Interrupted songs should be automatically dropped and next song in the queue should kick in immediately upon return to normal AutoDj operations.  Haven't seen any switch to change that behavior...

**Reply by PumpkinFM** [reply]
A little bit after the event but I have the same problems. The song playing before the top of the hour is paused then the TOTH sting plays the news and unfortunately the song playing before the TOTH continues afterwards, Driving me mental...
